### PR TITLE
[readability-js] Add article metadata on top of the page

### PR DESCRIPTION
**Pull request by rien333**
This PR solves two problems.

### Problem 1 ###

Unlike chromium's and firefox's reader mode, qute's reader output doesn't display the article's title on top of the page. This was done intentionally, because I felt that one could simply glance the title from the tab bar.

I changed my mind, however. First, one's tab bar can be  hidden. Second, it can also be too cramped to contain discernible textual information. Often, I found myself longing for a quick, easy to parse way to tell what a reader tab I once opened was all about. 

Given the screenshot below, it's difficult to tell what the focused article is about.
qute-wo-border

I think this PR improves this a lot:

qute1

It's also more aesthetically pleasing (imo), and more in line with what people expect from a "reader mode".  (note that the text justification is custom css enabled by the work done in https://github.com/qutebrowser/qutebrowser/commit/55fdae8eed0f93152b5676cca77b6243fd95d9b0)

---

### Problem 2

There's no way to easily get back to the original article if you happened to have closed it (which is not true for firefox/chromium, since they implement reader mode with a toggle button). Hence, I decided to credit the website source in the subtitle as a clickable link. 

mozilla's readability library often returns a website's source as a natural name (e.g. "Encyclopedia Britannica", see above). If this source is unavailable (i.e. `null`) , this PR shows the "base url" of the original site instead. 

For example:

qute2

### CODE REVIEW

**Comment by rien333** [comment]
Per comparison, this is what Firefox title heading looks like:
firefox

File: misc/userscripts/readability-js [hunk 1]
```diff
@@ -108,6 +117,9 @@ const HEADER = `
     yaDhjMS4xMSAwIDItMC44OTUgMi0ycy0wLjg5NS0yLTItMnoiIGZpbGw9IiNmZmYiLz4KPC9nPgo8L3N2Zz4K"/>
 </head>
 <body class="qute-readability">
+    <h1 class="reader-title">%s</h1>
+    <div>From <a class="reader-title" href=%s>%s</a></div>
```
Comment (The-Compiler) [inline misc/userscripts/readability-js:121]:
```suggestion
    <div>From <a class="reader-title" href="%s">%s</a></div>
```

I think?

**Review by The-Compiler — CHANGES_REQUESTED** [review]
This probably was an issue even before this change, but shouldn't those strings be escaped some way before inserting them into the HTML (or alternatively, use some kind of template engine)? Right now, if there was something like a `<` in a title, we'd insert that literally, no?

File: misc/userscripts/readability-js (hunk 1 above)
Comment (rien333) [reply misc/userscripts/readability-js:121]: Right, nice catch. Let me think about the best solution (e.g. a template engine).

Comment (The-Compiler) [reply misc/userscripts/readability-js:121]: I think the double quotes and the templating thing are two different issues. The double quotes are required no matter what for this to be valid HTML.

**Comment by The-Compiler** [comment]
Thanks! Also I really appreciate the screenshots :slightly_smiling_face:

**Comment by rien333** [comment]
> Also I really appreciate the screenshots

Good to hear, I was worried they only would add to my already lengthy PR. 

With respect to the template suggestion you raised above: I think you're right that there are more elegant/fool proof solutions than just inserting stuff using `util.format(...)` (though I would add that I use this userscript almost daily, and it never fails in some unexpected way). 

I was thinking about rewriting that part of the code anyway, since it would be cool if the python and javascript variants of this userscripts could somehow share the same html/css "template". I'm still thinking about how to do that though.

**Comment by The-Compiler** [comment]
Yeah, I guess there's a lot of value in keeping userscripts self-contained. IMHO I wouldn't worry too much about the Python one, seeing how it's definitely inferior to the JS one.

**Pull request merged**